api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -402,6 +402,10 @@
      */
     Attribute<Boolean> BUFFER_SPOOLING = new Attribute<Boolean>(true);
     /**
+     * Whether to apply back pressure and stall the upstream operator when the downstream operator is slow.
+     */
+    Attribute<Boolean> BUFFER_BACK_PRESSURE = new Attribute<Boolean>(true);
```

Comment:
I do see a use case where an operator may want to burst output, for example in a case of an aggregation: an operator waits to emit the result till the aggregation is done and while waiting, it may accumulate tuples in memory. Once the aggregation is done, the operator emits a large chunk of data that can and should be spooled to disk and later it will be consumed by subscribers. At the same time, the operator is ready to accumulate new tuples in memory for the next aggregation.

Putting the attribute on the DAG assumes that it is intended for preserving backward compatibility with the old behavior. To handle burst output, it should be an attribute of the stream (output port).